cnudie/util.py

```diff
@@ -292,3 +294,29 @@ def enumerate_group_pairs(
             _add_if_not_duplicate(resource_diff.resource_refs_only_right, i)
 
     return resource_diff
+
+
+CTF_CD_FILENAME = 'component-descriptor.yaml'
+
+
+def extract_component_descriptor_from_ctf_archive(
+    path_to_ctf_tarfile: str
+) -> cm.ComponentDescriptor:
+    '''Extract the component descriptor from a CTF-tarfile
```

Comment:
also, it is not essential for the contract of this function that it "extracts" the component-descriptor. what is essential is that it returns it to the caller